libcontainer/system/linux.go

```diff
@@ -215,3 +230,103 @@ func SetLinuxPersonality(personality int) error {
 	}
 	return nil
 }
+
+// UpdateSetgroups is to set the process's setgroups policy
+// This *must* be called before we touch gid_map.
+func UpdateSetgroups(pid int, policy SetgroupsPolicy) error {
+	var strPolicy string
+	switch policy {
+	case SetgroupsAllow:
+		strPolicy = "allow"
+	case SetgroupsDeny:
+		strPolicy = "deny"
+	case SetgroupsDefault:
+		fallthrough
+	default:
+		return nil
+	}
+	err := utils.WriteString(fmt.Sprintf("/proc/%d/setgroups", pid), strPolicy)
+	/*
+	 * If the kernel is too old to support /proc/pid/setgroups,
+	 * open(2) or write(2) will return ENOENT. This is fine.
+	 */
```

Comment:
nit: better to use `//` comments in go (I'm not quite sure why, guess it's implied coding style)